pkg/controller/sshd/sshd_controller.go

```diff
@@ -378,6 +452,45 @@ func newSSHDDeployment(cr *cloudingressv1alpha1.SSHD, configMapList *corev1.Conf
 			MountPath: filepath.Join(authorizedKeysMountPath, volumeName),
 		})
 	}
+	// Sort volume slices by name to keep the sequence stable.
+	sort.Slice(volumes, func(i, j int) bool {
+		return volumes[i].Name < volumes[j].Name
+	})
+	sort.Slice(volumeMounts, func(i, j int) bool {
+		return volumeMounts[i].Name < volumeMounts[j].Name
+	})
+
+	volumeName := hostKeysSecret.ObjectMeta.Name
+	volumes = append(volumes, corev1.Volume{
+		Name: volumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  hostKeysSecret.ObjectMeta.Name,
+				DefaultMode: pointer.Int32Ptr(0600),
+				Optional:    pointer.BoolPtr(false),
+			},
+		},
+	})
+	volumeMounts = append(volumeMounts, corev1.VolumeMount{
+		Name:      volumeName,
+		MountPath: hostKeysMountPath,
+	})
+
+	// Add environment variables to help the container configure itself.
+	var env []corev1.EnvVar
+	env = append(env, corev1.EnvVar{
+		Name:  "AUTHORIZED_KEYS_DIR",
+		Value: authorizedKeysMountPath,
+	})
+	for key := range hostKeysSecret.Data {
+		env = append(env, corev1.EnvVar{
+			Name:  strings.ToUpper(key),
+			Value: filepath.Join(hostKeysMountPath, key),
+		})
+	}
```

Comment:
What's happening here? There's only one element in the secret's data (cf lines 424-428), so why iterate over it like this instead of a direct reference to that single element?

Reply:
On the possibility that we might [support other key types](https://github.com/openshift/cloud-ingress-operator/pull/91/files#diff-e5ea7d858af86730d41fd21b37ca353aR409-R413).  Just an attempt at future proofing.

```
// XXX Generate other key types?
//     ECDSA:   Easy, fully supported in standard library.
//     ED25519: Standard library can generate a private key, but requires an
//              external module to create an "OPENSSH PRIVATE KEY" PEM block:
//              https://github.com/mikesmitty/edkey
```

Reply:
Ah, I see. Be a little cautious of premature optimizing. :)